[TEST] exporting/.../test_exporting_engine.h: error: unknown type name mongoc_uri_t

<!--
When creating a bug report please:
- Verify first that your issue is not already reported on GitHub.
- Test if the latest release and master branch are affected too.
-->

##### Bug report summary
netdata 1.23.0 has a failing test:
```
In file included from exporting/tests/test_exporting_engine.c:3:
exporting/tests/test_exporting_engine.h:161:1: error: unknown type name ‘mongoc_uri_t’
  161 | mongoc_uri_t *__wrap_mongoc_uri_new_with_error(const char *uri_string, bson_error_t *error);
      | ^~~~~~~~~~~~
exporting/tests/test_exporting_engine.h:161:72: error: unknown type name ‘bson_error_t’
```

Here's the full build output: https://729724.bugs.gentoo.org/attachment.cgi?id=646548
And here's the downstream bug report: https://bugs.gentoo.org/729724
##### OS / Environment
<!--
Provide as much information about your environment (which operating system and distribution you're using, if Netdata is running in a container, etc.)
as possible to allow us reproduce this bug faster.

To get this information, execute the following commands based on your operating system:
- uname -a; grep -Hv "^#" /etc/*release  # Linux
- uname -a; uname -K                     # BSD
- uname -a; sw_vers                      # macOS

Place the output from the command in the code section below.  
 -->
```
$ uname -a; grep -Hv "^#" /etc/*release
Linux irrational 5.4.33-gentoo-x86_64 #2 SMP Sat Apr 18 15:49:12 EDT 2020 x86_64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz GenuineIntel GNU/Linux
/etc/gentoo-release:Gentoo Base System release 2.7
/etc/lsb-release:DISTRIB_ID="Gentoo"
/etc/os-release:NAME=Gentoo
/etc/os-release:ID=gentoo
/etc/os-release:PRETTY_NAME="Gentoo/Linux"
/etc/os-release:ANSI_COLOR="1;32"
/etc/os-release:HOME_URL="https://www.gentoo.org/"
/etc/os-release:SUPPORT_URL="https://www.gentoo.org/support/"
/etc/os-release:BUG_REPORT_URL="https://bugs.gentoo.org/"
```

##### Netdata version
<!--
Provide output of `netdata -V`.
 
If Netdata is running, execute: $(ps aux | grep -E -o "[a-zA-Z/]+netdata ") -V
 -->
 1.23.0

##### Component Name
<!--
Let us know which component is affected by the bug. Our code is structured according to its component,
so the component name is the same as the top level directory of the repository.
For example, a bug in the dashboard would be under the web component.
-->
tests
##### Steps To Reproduce
<!--
Describe how you found this bug and how we can reproduce it, preferably with a minimal test-case scenario.
If you'd like to attach larger files, use gist.github.com and paste in links.
-->
Run tests using `make check`

##### Expected behavior
<!-- Provide a clear and concise description of what you expected to happen. -->
Tests should pass.

--- a/exporting/tests/test_exporting_engine.h
+++ b/exporting/tests/test_exporting_engine.h
@@ -116,6 +116,7 @@ int __mock_end_chart_formatting(struct instance *instance, RRDSET *st);
 int __mock_end_host_formatting(struct instance *instance, RRDHOST *host);
 int __mock_end_batch_formatting(struct instance *instance);
 
+#if ENABLE_PROMETHEUS_REMOTE_WRITE
 void *__real_init_write_request();
 void *__wrap_init_write_request();
 
@@ -137,7 +138,9 @@ void __wrap_add_metric(
     void *write_request_p,
     const char *name, const char *chart, const char *family, const char *dimension,
     const char *instance, const double value, const int64_t timestamp);
+#endif /* ENABLE_PROMETHEUS_REMOTE_WRITE */
 
+#if HAVE_KINESIS
 void __wrap_aws_sdk_init();
 void __wrap_kinesis_init(
     void *kinesis_specific_data_p, const char *region, const char *access_key_id, const char *secret_key,
@@ -146,7 +149,9 @@ void __wrap_kinesis_put_record(
     void *kinesis_specific_data_p, const char *stream_name, const char *partition_key, const char *data,
     size_t data_len);
 int __wrap_kinesis_get_result(void *request_outcomes_p, char *error_message, size_t *sent_bytes, size_t *lost_bytes);
+#endif /* HAVE_KINESIS */
 
+#if ENABLE_EXPORTING_PUBSUB
 int __wrap_pubsub_init(
     void *pubsub_specific_data_p, char *error_message, const char *destination, const char *credentials_file,
     const char *project_id, const char *topic_id);
@@ -156,7 +161,9 @@ int __wrap_pubsub_publish(
 int __wrap_pubsub_get_result(
     void *pubsub_specific_data_p, char *error_message,
     size_t *sent_metrics, size_t *sent_bytes, size_t *lost_metrics, size_t *lost_bytes);
+#endif /* ENABLE_EXPORTING_PUBSUB */
 
+#if HAVE_MONGOC
 void __wrap_mongoc_init();
 mongoc_uri_t *__wrap_mongoc_uri_new_with_error(const char *uri_string, bson_error_t *error);
 int32_t __wrap_mongoc_uri_get_option_as_int32(const mongoc_uri_t *uri, const char *option, int32_t fallback);
@@ -175,6 +182,7 @@ bool __wrap_mongoc_collection_insert_many(
     const bson_t *opts,
     bson_t *reply,
     bson_error_t *error);
+#endif /* HAVE_MONGOC */
 
 // -----------------------------------------------------------------------
 // fixtures

